File: sort-mapper-example-java/src/main/java/kh/farrukh/sortmapperexamplejava/lesson/model/LessonResponseDTO.java
package kh.farrukh.sortmapperexamplejava.lesson.model;

public record LessonResponseDTO(
    Long id,
    String title,
    String description,
    Long courseId,
    String courseTitle,
    String teacherFirstName,
    String teacherLastName,
    Long helperTeacherId,
    String helperTeacherFirstName,
    String helperTeacherLastName
) {

}

File: sort-mapper-example/src/main/kotlin/kh/farrukh/sortmapperexample/lesson/model/LessonResponseDTO.kt
package kh.farrukh.sortmapperexample.lesson.model

import kh.farrukh.sortmapper.annotation.SortField


data class LessonResponseDTO(
    val id: Long,
    val title: String,
    val description: String?,
    val courseId: Long,
    val courseTitle: String,
    @SortField("course.teacher.firstName")
    val teacherFirstName: String,
    @SortField("course.teacher.lastName")
    val teacherLastName: String,
    val helperTeacherId: Long?,
    val helperTeacherFirstName: String?,
    val helperTeacherLastName: String?
)
